### PR TITLE
Adjacency list feature

### PR DESCRIPTION
**Pull request by dnbss**

### CODE REVIEW

**Review by EugenKoulik — APPROVED** [review]

File: GraphRepresentation/AdjacencyList.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using CraphModel;
+using GraphModelDraw;
+
+namespace GraphRepresentation
+{
+    public class AdjacencyList
+    {
+        Dictionary<int, List<Node>> adjacencyList;
```
Comment (Maksim-Trolina) [inline GraphRepresentation/AdjacencyList.cs:10]: Use the Vertex class, not Node

File: GraphRepresentation/AdjacencyList.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            else
+
+            {
+                Node node = new Node();
+
+                node.Connectable = vertexEnd.Id;
+
+                node.Weight = weight;
+
+                adjacencyList[vertexStart.Id].Add(node);
```
Comment (Maksim-Trolina) [inline GraphRepresentation/AdjacencyList.cs]: You can write in 1 line

File: GraphRepresentationTest/AdjacencyListTest.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public class AdjacencyListTest
+    {
+
+        [Test]
+        public void AddVertexTest_AddExistVertex_ExceptionExpected()
+        {
+
+            List<Vertex> verticles = new List<Vertex>();
+            Vertex vertex = new Vertex();
+            verticles.Add(vertex);
```
Comment (Maksim-Trolina) [inline GraphRepresentationTest/AdjacencyListTest.cs]: You can write in 1 line

File: GraphRepresentationTest/AdjacencyListTest.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            Assert.AreEqual(ex.Message, "This vertex is already in the list of adjacencies");
+
+        }
+
+        [Test]
+        public void AddNodeTest_CreateLoop_ExceptionExpected()
+        {
+            List<Vertex> verticles = new List<Vertex>();
+            Vertex vertex = new Vertex();
+            verticles.Add(vertex);
```
Comment (Maksim-Trolina) [inline GraphRepresentationTest/AdjacencyListTest.cs]: You can write in 1 line

**Review by Maksim-Trolina — APPROVED** [review]

**Pull request merged**